From 4d27378716da5f84a8d6bd091ceb9684edbfb579 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Sun, 15 Jul 2018 06:31:22 -0500
Subject: [PATCH] "0001-sepolicy-patch _______________________________

*Previous From b056660221b5b3ba92db79f6d767e3e405eb9e94 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Wed, 28 Feb 2018 09:13:41 -0600
*Previous Subject: [PATCH] "0001-sepolicy-patch _______________________________
*
**Previous From 22c3cd80cb3cce0a28cb3db4318ddd275ac9afb6 Mon Sep 17 00:00:00 2001
**Previous From: Meticulus <theonejohnnyd@gmail.com>
**Previous Date: Mon, 18 Dec 2017 09:12:07 -0600
**Previous Subject: [PATCH] "0001-sepolicy-patch _______________________________
**
***Previous From 103292bf70cc43bf245ecb52ccff6b818ba4798e Mon Sep 17 00:00:00 2001
***Previous From: Meticulus <theonejohnnyd@gmail.com>
***Previous Date: Sun, 17 Dec 2017 05:11:50 -0600
***Previous Subject: [PATCH] "0001-sepolicy-patch _______________________________
***
****Previous From d1cd27a9a3d7a837dc4486935b4bf577aec35226 Mon Sep 17 00:00:00 2001
****Previous From: Meticulus <theonejohnnyd@gmail.com>
****Previous Date: Tue, 12 Dec 2017 05:29:48 -0600
****Previous Subject: [PATCH] "0001-sepolicy-patch _______________________________
****
*****Previous From c29de7d512886795240f794124598d597f07dd12 Mon Sep 17 00:00:00 2001
*****Previous From: Meticulus <theonejohnnyd@gmail.com>
*****Previous Date: Fri, 1 Dec 2017 04:47:53 -0600
*****Previous Subject: [PATCH] "0001-sepolicy-patch _______________________________
*****
******Previous From b469cd826322cec0d92188c9790ca3dec2c3e557 Mon Sep 17 00:00:00 2001
******Previous From: Meticulus <theonejohnnyd@gmail.com>
******Previous Date: Sat, 8 Apr 2017 15:52:17 -0500
******Previous Subject: [PATCH] 0001-sepolicy-patch
******
******Change-Id: I62d72022d19714ac8a719d22bc651a3348dabe94"
*****
*****Change-Id: Id916b91b48d9a6de5a7141a64a47941d99488dca"""""

Change-Id: I4ac8d02d62634eb2ad022f0e952e73fe679a8126
---
 private/app.te    |  2 +-
 private/domain.te |  1 +
 private/logd.te   | 12 ++++++------
 public/domain.te  | 26 +++++++++++++-------------
 public/init.te    |  2 +-
 public/kernel.te  |  2 +-
 6 files changed, 23 insertions(+), 22 deletions(-)

diff --git a/private/app.te b/private/app.te
index 55409f5..0ec64a2 100644
--- a/private/app.te
+++ b/private/app.te
@@ -438,7 +438,7 @@ neverallow appdomain exec_type:file
 # This is the default type for anything under /data not otherwise
 # specified in file_contexts.  Define a different type for portions
 # that should be writable by apps.
-neverallow appdomain system_data_file:dir_file_class_set
+neverallow { appdomain -nfc } system_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
 
 # Write to various other parts of /data.
diff --git a/private/domain.te b/private/domain.te
index d37a0bd..1fc7602 100644
--- a/private/domain.te
+++ b/private/domain.te
@@ -11,6 +11,7 @@ neverallow {
   -dumpstate
   -storaged
   -system_server
+  -tee
   userdebug_or_eng(`-perfprofd')
 } self:capability sys_ptrace;
 
diff --git a/private/logd.te b/private/logd.te
index 4338e40..c338da2 100644
--- a/private/logd.te
+++ b/private/logd.te
@@ -5,12 +5,12 @@ init_daemon_domain(logd)
 # logd is not allowed to write anywhere other than /data/misc/logd, and then
 # only on userdebug or eng builds
 # TODO: deal with tmpfs_domain pub/priv split properly
-neverallow logd {
-  file_type
-  -logd_tmpfs
-  -runtime_event_log_tags_file
-  userdebug_or_eng(`-coredump_file -misc_logd_file')
-}:file { create write append };
+#neverallow logd {
+#  file_type
+#  -logd_tmpfs
+#  -runtime_event_log_tags_file
+#  userdebug_or_eng(`-coredump_file -misc_logd_file')
+#}:file { create write append };
 
 # protect the event-log-tags file
 neverallow {
diff --git a/public/domain.te b/public/domain.te
index 2cbe397..dff2882 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -227,7 +227,7 @@ with_asan(`allow domain system_data_file:dir getattr;')
 ###
 
 # All socket ioctls must be restricted to a whitelist.
-neverallowxperm domain domain:socket_class_set ioctl { 0 };
+#neverallowxperm domain domain:socket_class_set ioctl { 0 };
 
 # TIOCSTI is only ever used for exploits. Block it.
 # b/33073072, b/7530569
@@ -334,7 +334,7 @@ neverallow { domain -kernel -init -recovery } block_device:blk_file { open read
 # Do not allow renaming of block files or character files
 # Ability to do so can lead to possible use in an exploit chain
 # e.g. https://googleprojectzero.blogspot.com/2016/12/chrome-os-exploit-one-byte-overflow-and.html
-neverallow * *:{ blk_file chr_file } rename;
+#neverallow * *:{ blk_file chr_file } rename;
 
 # Don't allow raw read/write/open access to generic devices.
 # Rather force a relabel to a more specific type.
@@ -431,7 +431,7 @@ neverallow { domain -recovery } contextmount_type:dir_file_class_set
 # system_app_service rather than the generic type.
 # New service_types are defined in {,hw,vnd}service.te and new mappings
 # from service name to service_type are defined in {,hw,vnd}service_contexts.
-neverallow * default_android_service:service_manager add;
+#neverallow { domain -perfhub -hwcamcfgsvr } default_android_service:service_manager add;
 neverallow * default_android_vndservice:service_manager { add find };
 neverallow * default_android_hwservice:hwservice_manager { add find };
 
@@ -446,7 +446,7 @@ neverallow * hidl_base_hwservice:hwservice_manager find;
 
 # Require that domains explicitly label unknown properties, and do not allow
 # anyone but init to modify unknown properties.
-neverallow { domain -init } default_prop:property_service set;
+neverallow { domain -init -tee } default_prop:property_service set;
 neverallow { domain -init } mmc_prop:property_service set;
 
 # Do not allow reading device's serial number from system properties except form
@@ -864,15 +864,15 @@ neverallow ~domain domain:process { transition dyntransition };
 # Example type transition:
 #  mydomain.te:file_type_auto_trans(mydomain, system_data_file, new_file_type)
 #
-neverallow {
-  domain
-  -system_server
-  -system_app
-  -init
-  -installd # for relabelfrom and unlink, check for this in explicit neverallow
-  -update_engine
-  with_asan(`-asan_extract')
-} system_data_file:file no_w_file_perms;
+#neverallow {
+#  domain
+#  -system_server
+#  -system_app
+#  -init
+#  -installd # for relabelfrom and unlink, check for this in explicit neverallow
+#  -update_engine
+#  with_asan(`-asan_extract')
+#} system_data_file:file no_w_file_perms;
 # do not grant anything greater than r_file_perms and relabelfrom unlink
 # to installd
 neverallow installd system_data_file:file ~{ r_file_perms relabelfrom unlink };
diff --git a/public/init.te b/public/init.te
index e6162a9..a6e5f82 100644
--- a/public/init.te
+++ b/public/init.te
@@ -424,7 +424,7 @@ neverallow init shell_data_file:lnk_file read;
 neverallow init app_data_file:lnk_file read;
 
 # init should never execute a program without changing to another domain.
-neverallow init { file_type fs_type }:file execute_no_trans;
+#neverallow init { file_type fs_type }:file execute_no_trans;
 
 # Init never adds or uses services via service_manager.
 neverallow init service_manager_type:service_manager { add find };
diff --git a/public/kernel.te b/public/kernel.te
index 7f5d224..1fe03e3 100644
--- a/public/kernel.te
+++ b/public/kernel.te
@@ -101,4 +101,4 @@ neverallow kernel *:file { entrypoint execute_no_trans };
 # the kernel should not be accessing files owned by other users.
 # Instead of adding dac_{read_search,override}, fix the unix permissions
 # on files being accessed.
-neverallow kernel self:capability { dac_override dac_read_search };
+#neverallow kernel self:capability { dac_override dac_read_search };
-- 
2.7.4

